external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -689,57 +691,6 @@ abstract class KafkaMicroBatchSourceSuiteBase extends KafkaSourceSuiteBase {
     )
   }
 
-  test("allow group.id prefix") {
```

Comment:
No need the tests because driver is not user `KafkaConsumer` => no GroupId needed.

Reply:
Let's remove them from the doc as well. (There should be configuration as well as explanation around group id.)

Reply:
Executors still take advantage of this. How do you mean remove?

Reply:
Ah OK I missed. We can still remove the configuration for "exact" assign of groupId (`kafka.group.id`).

Btw, given the executor still leverage the config, can the removed tests be used to verify the executor side?

Reply:
Related removing "exact" assign of groupId (`kafka.group.id`) not sure what you mean. GroupId can be overwritten on executor side too. Please see (generated GroupId is set only if not overriden): https://github.com/apache/spark/blob/99384d1e831b7fe82a3a80ade1da976971624ee7/external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala#L609

I've recalled and from test perspective now I remember what was the situation. Originally there were no tests which covered prefix + override features. I've added the tests which covered driver side but the approach was not working for executors so not added such test. This still stands so unless somebody has an idea we can't use this approach to cover executor side.

Reply:
Ah OK, executors also can use fixed group id. My bad. I missed.

Reply:
OK, resolving this comment.